src/BCrypt.Net/BCrypt.cs

```diff
@@ -606,7 +606,7 @@ public static string HashPassword(string inputKey, string salt, bool enhancedEnt
 
             if (string.IsNullOrEmpty(salt))
```

Comment:
@ChrisMcKee should the check be `IsNullOrWhiteSpace` in that case?

Reply:
IsNullOrWhiteSpace was a convenience method added in .net 4; we still target 2.0 upwards / not much to gain from adding the extra checks as they salt would still fail at the next step.
The checks at the top are only really there as a basic guard statement.